tests/test_urls.py

```diff
@@ -293,3 +299,46 @@ def test_custom_404_rst_source(app, status, warning):
 
     for chunk in chunks:
         assert chunk in content
+
+
+@pytest.mark.sphinx(
+    srcdir=srcdir,
+)
+def test_toctree_links(app, status, warning):
+    app.build()
+    path = app.outdir / '404.html'
```

Comment:
are these tests running with py2? py2 doesn't have pathlib

Reply:
Yes, they are running with Py2. Although, this is Sphinx magic using an internal `abspath` util.